src/error.rs

```diff
@@ -0,0 +1,7 @@
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum Error {
+    #[error("{0} is not a valid absolute IRI")]
+    InvalidURIError(String), // TODO: is it possible to accept both the string _and_ a #[from] parameter? does this even make sense?
```

Comment:
No, you can only use either a parameter with the `#[from]` attribute, *or* a different parameter. From the `thiserror` docs:
> A From impl is generated for each variant containing a #[from] attribute.
>
> Note that the variant must not contain any other fields beyond the source error and possibly a backtrace. A backtrace is captured from within the From impl if there is a field for it.